### ISSUE TITLE
 ./env/install-packt-beam.sh fails on CentOS7

### ISSUE THREAD

**Issue by donhk** [PacktPublishing/Building-Big-Data-Pipelines-with-Apache-Beam#27]
```
[donhk@workpc Building-Big-Data-Pipelines-with-Apache-Beam]$ ./env/install-packt-beam.sh
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: linux
[INFO] os.detected.arch: x86_64
[INFO] os.detected.bitness: 64
[INFO] os.detected.version: 3.10
[INFO] os.detected.version.major: 3
[INFO] os.detected.version.minor: 10
[INFO] os.detected.release: centos
[INFO] os.detected.release.version: 7
[INFO] os.detected.release.like.centos: true
[INFO] os.detected.release.like.rhel: true
[INFO] os.detected.release.like.fedora: true
[INFO] os.detected.classifier: linux-x86_64
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] Building Big Data Pipelines with Apache Beam                       [pom]
[INFO] Building Big Data Pipelines with Apache Beam Utils                 [jar]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 1             [jar]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 2             [jar]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 3             [jar]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 4             [jar]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 5             [jar]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 6             [jar]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 7             [jar]
[INFO]
[INFO] -------------------< com.packtpub.beam:beam-parent >--------------------
[INFO] Building Building Big Data Pipelines with Apache Beam 1.0.0        [1/9]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- spotless-maven-plugin:2.0.1:apply (apply) @ beam-parent ---
[INFO]
[INFO] --------------------< com.packtpub.beam:beam-util >---------------------
[INFO] Building Building Big Data Pipelines with Apache Beam Utils 1.0.0  [2/9]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]

...
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ beam-chapter6 ---
[INFO] No tests to run.
[INFO]
[INFO] --- exec-maven-plugin:3.0.0:exec (default) @ beam-chapter6 ---
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/1.5.4/plexus-component-annotations-1.5.4.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/1.5.4/plexus-component-annotations-1.5.4.pom (815 B at 14 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-containers/1.5.4/plexus-containers-1.5.4.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-containers/1.5.4/plexus-containers-1.5.4.pom (4.2 kB at 80 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus/2.0.5/plexus-2.0.5.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus/2.0.5/plexus-2.0.5.pom (17 kB at 369 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.0.20/plexus-utils-3.0.20.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.0.20/plexus-utils-3.0.20.pom (3.8 kB at 73 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/1.6/plexus-component-annotations-1.6.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/1.6/plexus-component-annotations-1.6.pom (748 B at 9.6 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-containers/1.6/plexus-containers-1.6.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-containers/1.6/plexus-containers-1.6.pom (3.8 kB at 75 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus/3.3.2/plexus-3.3.2.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus/3.3.2/plexus-3.3.2.pom (22 kB at 422 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.pom (11 kB at 256 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-parent/35/commons-parent-35.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-parent/35/commons-parent-35.pom (58 kB at 466 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/apache/15/apache-15.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/apache/15/apache-15.pom (15 kB at 287 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-artifact-transfer/0.10.1/maven-artifact-transfer-0.10.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-artifact-transfer/0.10.1/maven-artifact-transfer-0.10.1.pom (11 kB at 279 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.pom (4.8 kB at 60 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-utils/3.1.0/maven-shared-utils-3.1.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-utils/3.1.0/maven-shared-utils-3.1.0.pom (5.0 kB at 59 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/slf4j/slf4j-api/1.7.5/slf4j-api-1.7.5.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/slf4j/slf4j-api/1.7.5/slf4j-api-1.7.5.pom (2.7 kB at 60 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/slf4j/slf4j-parent/1.7.5/slf4j-parent-1.7.5.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/slf4j/slf4j-parent/1.7.5/slf4j-parent-1.7.5.pom (12 kB at 281 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.0.20/plexus-utils-3.0.20.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/1.6/plexus-component-annotations-1.6.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-artifact-transfer/0.10.1/maven-artifact-transfer-0.10.1.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.0.20/plexus-utils-3.0.20.jar (243 kB at 1.9 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-utils/3.1.0/maven-shared-utils-3.1.0.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/1.6/plexus-component-annotations-1.6.jar (4.3 kB at 30 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/slf4j/slf4j-api/1.7.5/slf4j-api-1.7.5.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.jar (61 kB at 384 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar (54 kB at 267 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-utils/3.1.0/maven-shared-utils-3.1.0.jar (164 kB at 810 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-artifact-transfer/0.10.1/maven-artifact-transfer-0.10.1.jar (128 kB at 584 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/slf4j/slf4j-api/1.7.5/slf4j-api-1.7.5.jar (26 kB at 110 kB/s)
Collecting apache-beam==2.33.0 (from -r requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/d1/69/1ee72105f67c12ee82aa2c989d17ec58923cf07984e5af077ddf60d69022/apache_beam-2.33.0-cp36-cp36m-manylinux1_x86_64.whl (9.8MB)
Collecting protobuf==3.17.3 (from -r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/53/4e/e2db88d0bb0bda6a879eea62fddbaf813719ce3770d458bc5580512d9c95/protobuf-3.17.3-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.0MB)
Collecting grpcio==1.40.0 (from -r requirements.txt (line 4))
  Downloading https://files.pythonhosted.org/packages/37/56/700ff83b59eb69a7ddd62840666699173923020cfdf33a2860b9eba5667a/grpcio-1.40.0.tar.gz (21.3MB)
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-wn4f9rq9/grpcio/setup.py", line 255, in <module>
        if check_linker_need_libatomic():
      File "/tmp/pip-build-wn4f9rq9/grpcio/setup.py", line 205, in check_linker_need_libatomic
        stderr=PIPE)
      File "/usr/lib64/python3.6/subprocess.py", line 729, in __init__
        restore_signals, start_new_session)
      File "/usr/lib64/python3.6/subprocess.py", line 1364, in _execute_child
        raise child_exception_type(errno_num, err_msg, err_filename)
    FileNotFoundError: [Errno 2] No such file or directory: 'c++': 'c++'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-wn4f9rq9/grpcio/
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:982)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:929)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:457)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Building Big Data Pipelines with Apache Beam 1.0.0:
[INFO]
[INFO] Building Big Data Pipelines with Apache Beam ....... SUCCESS [  0.449 s]
[INFO] Building Big Data Pipelines with Apache Beam Utils . SUCCESS [  5.787 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 1 SUCCESS [  4.422 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 2 SUCCESS [  5.882 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 3 SUCCESS [ 25.998 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 4 SUCCESS [  8.157 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 5 SUCCESS [ 10.602 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 6 FAILURE [ 15.358 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 7 SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:16 min
[INFO] Finished at: 2022-10-08T13:51:32-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.0.0:exec (default) on project beam-chapter6: Command execution failed.: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :beam-chapter6

```

 java -version
java version "11.0.16" 2022-07-19 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.16+11-LTS-199)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.16+11-LTS-199, mixed mode)

16GB RAM
8CPUs
minikube 6CPUs 8000M 

Looks like the problem comes from Python

**Comment by donhk** [comment]
As a workaround, I removed chapters 6 and 7 from the pom.xml but the build did not create the expected `packt-beam-*`

```
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ beam-chapter5 ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Building Big Data Pipelines with Apache Beam 1.0.0:
[INFO]
[INFO] Building Big Data Pipelines with Apache Beam ....... SUCCESS [  0.237 s]
[INFO] Building Big Data Pipelines with Apache Beam Utils . SUCCESS [  6.356 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 1 SUCCESS [  4.290 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 2 SUCCESS [  6.862 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 3 SUCCESS [ 24.108 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 4 SUCCESS [  7.681 s]
[INFO] Building Big Data Pipelines with Apache Beam Chapter 5 SUCCESS [  9.653 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  59.492 s
[INFO] Finished at: 2022-10-08T14:17:27-05:00
[INFO] ------------------------------------------------------------------------
Sending build context to Docker daemon   15.3MB
Step 1/5 : FROM packt-beam-base:latest
 ---> 9483a1c835ec
Step 2/5 : ADD */target/beam-*.jar /usr/local/lib/
 ---> 1057bfda05ae
Step 3/5 : ADD chapter*/src/main/python/* /usr/local/bin/
 ---> 35342fdf0667
Step 4/5 : ADD chapter6/src/main/python/dist/*.tar.gz /usr/local/bin/
ADD failed: no source files were specified
[donhk@workpc Building-Big-Data-Pipelines-with-Apache-Beam]$ kubectl get pods
NAME      READY   STATUS    RESTARTS      AGE
kafka-0   1/1     Running   1 (61m ago)   61m
zk-0      1/1     Running   0             61m
```

**Comment by donhk** [comment]
fixed by https://github.com/PacktPublishing/Building-Big-Data-Pipelines-with-Apache-Beam/pull/29

**Comment by maneesh299** [comment]
@donhk  so the fix is using python 3.7  which was created using a venv?

**Comment by donhk** [comment]
> [@donhk](https://github.com/donhk) so the fix is using python 3.7 which was created using a venv?

this was [the fix](https://github.com/PacktPublishing/Building-Big-Data-Pipelines-with-Apache-Beam/pull/29) which I believe was fixed later
